jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -47,6 +47,18 @@ JERRY_STATIC_ASSERT ((int32_t) ECMA_STRING_MAX_CONCATENATION_LENGTH ==
                      ECMA_STRING_MAX_CONCATENATION_LENGTH_should_be_representable_with_int32_t);
 
 /**
+ * The ecma string ref counter should start after the container field.
+ */
+JERRY_STATIC_ASSERT (ECMA_STRING_CONTAINER_MASK + 1 == ECMA_STRING_REF_ONE,
+                     ecma_string_ref_counter_should_start_after_the_container_field);
+
+/**
+ * The ecma string ref and container fields should fill the 16 bit field.
+ */
+JERRY_STATIC_ASSERT ((ECMA_STRING_MAX_REF | ECMA_STRING_CONTAINER_MASK) == UINT16_MAX,
+                     ecma_string_ref_and_container_fields_should_fill_the_16_bit_field);
```

Comment:
I still would like to see an assert and also an automatically calculated __COUNT label for the enum. But that does not necessarily require the extension of switches. 

I've looked for the switches with `ECMA_STRING_CONTAINER_` case labels and almost all of them already had default branches with some kind of an assert (unreachable, or checking for an explicitly given container type), which is what they should do IMHO. The only exception I've found is in function `ecma_deref_ecma_string`, which does not check for invalid values -- but it should (or already should have) IMHO!

Where have you found switches that needed extension?